proto/ffi/ffi.proto

```diff
@@ -0,0 +1,152 @@
+syntax = "proto3";
```

Comment:
can we not move the FFI proto files here? They don't really belong in this repo. I know they do depend on `api.proto`, but it'd be best to see if we could keep the FFI generated files out of here.